app/vmselect/netstorage/netstorage.go

```diff
@@ -353,11 +353,11 @@ func (rss *Results) runParallel(qt *querytracer.Tracer, f func(rs *Result, worke
 	for i := range workChs {
 		wg.Add(1)
 		qtChild := qt.NewChild("worker #%d", i)
-		go func(workerID uint) {
+		go func(workerID uint, qtChild *querytracer.Tracer) {
 			timeseriesWorker(qtChild, workChs, workerID)
 			qtChild.Done()
 			wg.Done()
-		}(uint(i))
+		}(uint(i), qtChild)
```

Comment:
According to my tests, it seems that not the root cause. 
Most likely related code maybe:
- https://github.com/VictoriaMetrics/VictoriaMetrics/blob/v1.93.7-cluster/app/vmselect/netstorage/netstorage.go#L1700-L1706
- https://github.com/VictoriaMetrics/VictoriaMetrics/blob/v1.93.7-cluster/app/vmselect/netstorage/netstorage.go#L1790-L1800

With `skipSlowReplicas` set, `collectResults` will not wait all storage node result. When marshaling the query tracer, the later storage result will be received and update query tracer concurrently, which caused data race, and maybe panic.

Reply:
Yes, marshaling causes this data race most likely. Need to properly fix it.